mteb/models/muq_mulan_model.py

```diff
@@ -0,0 +1,250 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+from functools import partial
+from typing import Any
+
+import numpy as np
+import torch
+import torchaudio
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from mteb.encoder_interface import AudioBatch, AudioData, PromptType
+from mteb.model_meta import ModelMeta
+from mteb.requires_package import requires_package
+
+
+class MuQMuLanWrapper:
+    def __init__(
+        self,
+        model_name: str = "OpenMuQ/MuQ-MuLan-large",
+        device: str = "cuda" if torch.cuda.is_available() else "cpu",
+        **kwargs: Any,
+    ):
+        requires_package(self, "muq", "pip install 'mteb[muq]'")
+        from muq import MuQMuLan
+
+        self.model_name = model_name
+        self.device = device
+        self.target_sampling_rate = 24000
+
+        # Load the model
+        self.model = MuQMuLan.from_pretrained(model_name).eval().to(self.device)
+
+    def _process_audio(self, audio: AudioBatch) -> list[torch.Tensor]:
+        """Process audio batch and return list of tensors."""
+        processed_audio = []
+
+        if isinstance(audio, DataLoader):
+            for batch in audio:
+                processed_audio.extend(self._handle_batch(batch))
+        else:
+            processed_audio = self._handle_batch(audio)
+
+        return processed_audio
+
+    def _handle_batch(
+        self, batch: AudioData | Iterable[tuple[AudioData, str]]
+    ) -> list[torch.Tensor]:
+        """Handle a single batch of audio data."""
```

Comment:
Seeing a lot of repeated code with the other PRs. Maybe we can extract and reuse this in `mteb/models/audio_utils.py` or something.